capstone/src/main/java/com/google/sps/servlets/StudySetServlet.java

```diff
@@ -362,4 +368,28 @@ private boolean areFieldsFilled(
     }
     return true;
   }
+
+  private String getOwnerId(DataSource pool) {
+    UserService userService = UserServiceFactory.getUserService();
+    Boolean isUserLoggedIn = userService.isUserLoggedIn();
+    if (!isUserLoggedIn) {
+      System.err.println("Need to be logged in");
+      return "";
```

Comment:
Returning empty values or null when something goes wrong is an anti-pattern. Just means your callers have to remember to always check whether the output is empty, easy to forget. Also you want to return a 401 Unauthorized error code to the user rather than silently printing to STDERR. I would suggest creating a custom exception, e.g. UnauthorizedException, and throw it here: https://www.baeldung.com/java-new-custom-exception#custom-unchecked. Then in the top-level doGet/doPost/etc catch that exception and set the response code to 401.